pysr/regressor_sequence.py

```diff
@@ -0,0 +1,192 @@
+            "Recursive symbolic regression only supports up to 2D data; please flatten your data first"
+        )
+    elif len(X) < 2:
+        raise ValueError(
+            "Recursive symbolic regression requires at least 2 datapoints; if you tried to pass a 1D array, use array.reshape(-1, 1)"
+        )
+    if len(X) <= recursive_history_length + 1:
+        raise ValueError(
+            f"Recursive symbolic regression with a history length of {recursive_history_length} requires at least {recursive_history_length + 2} datapoints."
+        )
+    if isinstance(weights, np.ndarray) and len(weights) != len(X):
+        raise ValueError("The length of `weights` must have shape (n_times,).")
+    if isinstance(variable_names, list) and len(variable_names) != X.shape[1]:
+        raise ValueError(
+            "The length of `variable_names` must be equal to the number of features in `X`."
+        )
+    if isinstance(X_units, list) and len(X_units) != X.shape[1]:
+        raise ValueError(
+            "The length of `X_units` must be equal to the number of features in `X`."
+        )
+    if y is not None:
+        raise ValueError("Recursive symbolic regression does not use `y`")
+    if y_units is not None:
+        raise ValueError("Recursive symbolic regression does not use `y_units`")
+    if Xresampled is not None:
+        raise ValueError("Recursive symbolic regression does not use `Xresampled`")
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.recursive_history_length = recursive_history_length
+
+    def _variable_names(self, y, variable_names=None):
+        if not variable_names:
+            if y.shape[1] == 1:
+                return [f"xt_{i}" for i in range(self.recursive_history_length, 0, -1)]
+            else:
+                return [
+                    f"x{i}t_{j}"
+                    for j in range(self.recursive_history_length, 0, -1)
+                    for i in range(y.shape[1])
+                ]
+        else:
+            return [
+                i + "t_" + str(j)
+                for j in range(self.recursive_history_length, 0, -1)
+                for i in variable_names
+            ]
+
+    def fit(
+        self,
+        X,
+        y=None,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+        y_units=None,
+    ) -> "PySRSequenceRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_times, n_features).
```

Comment:
Might also be nice to allow input of shape `(n_times,)`. This isn't allowed in the regular `PySRRegressor`, but only because a user might pass a single row of data with multiple features. But here, there is no ambiguity: you would always want to pass multiple times. So we can allow 1D input.

Reply:
Although I guess `self._validate_data` might require 2D... Hm...

Reply:
if it's 1D, can we just make it 2D with reshape(-1, 1) before calling _validate_data?